scaleway/resource_k8s_cluster_beta.go

```diff
@@ -480,38 +477,49 @@ func resourceScalewayK8SClusterBetaDefaultPoolUpdate(d *schema.ResourceData, m i
 	if d.HasChange("default_pool") {
 		defaultPoolID := d.Get("default_pool.0.pool_id").(string)
 
-		updateRequest := &k8s.UpdatePoolRequest{
-			Region: region,
-			PoolID: expandID(defaultPoolID),
-		}
+		forceNew := false
+		oldPoolID := ""
+		if d.HasChange("default_pool.0.container_runtime") || d.HasChange("default_pool.0.node_type") || d.HasChange("default_pool.0.placement_group_id") {
+			forceNew = true
+			oldPoolID = defaultPoolID
+		} else {
+			updateRequest := &k8s.UpdatePoolRequest{
+				Region: region,
+				PoolID: expandID(defaultPoolID),
+			}
 
-		if autohealing, ok := d.GetOk("default_pool.0.autohealing"); ok {
-			updateRequest.Autohealing = scw.BoolPtr(autohealing.(bool))
-		}
+			if autohealing, ok := d.GetOk("default_pool.0.autohealing"); ok {
+				updateRequest.Autohealing = scw.BoolPtr(autohealing.(bool))
+			}
 
-		if minSize, ok := d.GetOk("default_pool.0.min_size"); ok {
-			updateRequest.MinSize = scw.Uint32Ptr(uint32(minSize.(int)))
-		}
+			if minSize, ok := d.GetOk("default_pool.0.min_size"); ok {
+				updateRequest.MinSize = scw.Uint32Ptr(uint32(minSize.(int)))
+			}
 
-		if maxSize, ok := d.GetOk("default_pool.0.max_size"); ok {
-			updateRequest.MaxSize = scw.Uint32Ptr(uint32(maxSize.(int)))
-		}
+			if maxSize, ok := d.GetOk("default_pool.0.max_size"); ok {
+				updateRequest.MaxSize = scw.Uint32Ptr(uint32(maxSize.(int)))
+			}
 
-		if autoscaling, ok := d.GetOk("default_pool.0.autoscaling"); ok {
-			updateRequest.Autoscaling = scw.BoolPtr(autoscaling.(bool))
-		}
+			if autoscaling, ok := d.GetOk("default_pool.0.autoscaling"); ok {
+				updateRequest.Autoscaling = scw.BoolPtr(autoscaling.(bool))
+			}
 
-		if d.Get("default_pool.0.autoscaling").(bool) == false {
-			if size, ok := d.GetOk("default_pool.0.size"); ok {
-				updateRequest.Size = scw.Uint32Ptr(uint32(size.(int)))
+			if d.Get("default_pool.0.autoscaling").(bool) == false {
+				if size, ok := d.GetOk("default_pool.0.size"); ok {
+					updateRequest.Size = scw.Uint32Ptr(uint32(size.(int)))
+				}
 			}
-		}
 
-		_, err := k8sAPI.UpdatePool(updateRequest)
-		if err != nil {
-			if !is404Error(err) {
-				return err
+			_, err := k8sAPI.UpdatePool(updateRequest)
+			if err != nil {
+				if !is404Error(err) {
```

Comment:
if this is a 404 something is wrong

Reply:
either force recreate, either error

Reply:
Not it's desired behaviour. It was there before. If the default pool is deleted by hand, it should be recreated